src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -64,55 +62,28 @@
  * about these suffixes are added.
  */
 public class SkroutzGreekStemmer {
-  public final static String DEFAULT_STOPWORD_FILE = "stopwords.txt";
-  private final CharArraySet stopwords;
-  private Map<String, char[]> stepZeroExceptions;
+  private CharArraySet protectedWords = CharArraySet.EMPTY_SET;
+  private CharArrayMap stepZeroExceptions = CharArrayMap.emptyMap();
 
-  public SkroutzGreekStemmer(final CharArraySet stopwords) {
-    this.stopwords = stopwords;
+  public SkroutzGreekStemmer() {
     try {
-      this.stepZeroExceptions = loadExceptions("step_0_exceptions");
+      this.protectedWords = new CharArraySet(
+              loadProtected("protected_words"), true);
+      this.stepZeroExceptions = new CharArrayMap(
+              loadExceptions("step_0_exceptions"), true);
+
     } catch(Exception ex) {
       System.err.println("Failed to load exceptions from yaml file");
-      ex.printStackTrace();
-      this.stepZeroExceptions =  new HashMap(0);
```

Comment:
We used to initialize our data structures with null and therefore this line was necessary to handle the exceptions gracefully. However Lucene's CharArraySet & CharArrayMap offer this "empty set" and "empty map" so in case we won't load anything (exception) they are going to be empty - exactly what we were doing before.